src/test/java/net/sf/jabref/logic/groups/GroupTreeNodeTest.java

```diff
@@ -174,4 +187,47 @@ public void getSearchRuleForIncludingGroupReturnsGroupOrSubgroupAsMatcher() thro
         matcher.addRule(child.getGroup());
         assertEquals(matcher, node.getSearchRule());
     }
+
+    @Test
+    public void numberOfHitsReturnsZeroForEmptyList() throws Exception {
+        assertEquals(0, getNodeInSimpleTree().numberOfHits(Collections.emptyList()));
+    }
+
+    @Test
+    public void numberOfHitsMatchesOneEntry() throws Exception {
+        GroupTreeNode parent = getNodeInSimpleTree();
+        GroupTreeNode node = parent.addSubgroup(
+                new KeywordGroup("node", "author", "author2", true, false, GroupHierarchyType.INDEPENDENT));
+        assertEquals(1, node.numberOfHits(entries));
+    }
+
+    @Test
+    public void numberOfHitsMatchesMultipleEntries() throws Exception {
+        GroupTreeNode parent = getNodeInSimpleTree();
+        GroupTreeNode node = parent.addSubgroup(
+                new KeywordGroup("node", "author", "author1", true, false, GroupHierarchyType.INDEPENDENT));
+        assertEquals(2, node.numberOfHits(entries));
+    }
+
+    // Test for #1482
```

Comment:
Unnecessary comment in my opinion. Doesn't matter for which issue this test is. The test covers the old problem.